teps/0004-tekton-integrations-annotations-statuses.md

```diff
@@ -0,0 +1,689 @@
+caught at the time of initial validation.
+
+#### Why didn't this Run update my PR?
+
+With integration statuses, we have a place to annotate data alongside the
+pipeline run itself. This gives us an easy mechanism for users to do the
+following:
+
+1. Push a commit, notice pull request wasn't updated for a run.
+2. Look for all Runs with Integration annotation for the repo / commit (if
+   empty, then you know there was no build kicked off).
+3. Find latest Run, `kubectl get` it, inspect integrations status field(s) for
+   why it the status didn't appear.
+
+#### Multitentant / root secrets
+
+Some integrations like require shared secrets that aren't appropriate to be
+shared with users in a multitenant environment since they may grant broad
+access. For example, GitHub Apps authenticate by generating a token from a
+private key. If you have access to this private key, you can access any
+installation of the App across any user/org with the permissions of the App.
+
+Separating pipeline execution from integration handling allows for integrators
+to have more control over how and where these types of secrets are used,
+independent of general user workloads. As long as you have the ability to read
+integration annotations and update the integration status, you have the ability
+to run the actual integration handling any where (e.g. different cluster, or
+even outside of kubernetes altogether) and have control over the authorization
+with external systems.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+#### Integration Event Handling
+
+A key aspect of the proposal is that user Pipelines are not directly responsible
+for handling integration events, and mainly focuses on how integrations can
+configure itself and provide information back to users within the Run. It
+intentionally does not define how integrations should handle events.
+
+It is up to individual integrations to decide how and when to handle integration
+logic. Handlers should be installed independently of the core Pipeline
+installation to allow users to customize their installation.
+
+As a recommendation, integrations should rely on some form of reconciler in
+order to reliably process events, but it's also fine to rely on a push
+mechanism, or even rely on PipelineTasks to handle and annotate integration
+status data.
+
+#### Annotation-less integrations
+
+Defining annotations is an option, not a requirement. If you want an integration
+to apply to all Runs with no configuration needed (or provided by another source
+like a ConfigMap), that is okay. Integrations of this type will still able to
+use status fields.
+
+This is useful for use cases like webhook/pubsub notifications for all Run
+events.
```

Comment:
How does this "work" ? As what differs from "today" ? I guess "writing the status" is the only difference.